### ISSUE TITLE
Sandmark breaks on OCaml 4.14.0+trunk

### ISSUE THREAD

**Issue by kayceesrk** [ocaml-bench/sandmark#247]
See the failure in the CI at https://cloud.drone.io/ocaml-bench/sandmark/840/1/2. It would be useful to keep the CI green in the next few months as we head towards upstreaming multicore.

**Comment by kayceesrk** [comment]
@shubhamkumar13 This is high-priority for the compiler team. Please let's work on this.

**Comment by shubhamkumar13** [comment]
Yes I did looked into it yesterday and the issue seemed to be the dune variant.
The error log did mention it
```
<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build dune 2.8.1
+- 
```
But updating the dune version didn't fix it, there seem to be a lot of packages that might need updating and didn't build with trunk which I'm planning to look at. I have the error log [here](https://gist.github.com/shubhamkumar13/bacadeb95829211ad84c17532e49e1b9).

**Comment by kayceesrk** [comment]
Many packages are not expected to work with trunk. This may be due to several reasons. (1) It may be that trunk introduces a breaking change that makes the package not build with trunk. (2) It may be that the upper limit of the constraint is less than trunk's version; trunk always bumps the version number after a release.

**Comment by shubhamkumar13** [comment]
I tried to update the dependency packages to make sandmark work with 4.14.0+trunk.
There are still some of the packages which are still not working with 4.14.0+trunk.
```
react
rresult
fmt
astring
mtime
ocaml-migrate-parsetree
```

The packages `react, rresult, fmt, astring` have the same error related to `Toploop.use_file` .
There seems to be changes in the toploop interface (found a similar issue related to tooploop in https://github.com/ocaml/ocaml/pull/10461#issuecomment-872799250)

I also tested my branch other ci workflows locally and it seems to work without any issues.
